File: Data Structures & Algorithms/minimum-size-subarray-sum/submission-3.java
class Solution {
    public int minSubArrayLen(int target, int[] nums) {
        int left = 0, windowSub = 0, res = Integer.MAX_VALUE;
        int n = nums.length;
        for (int right = 0; right < n; right++) {

            windowSub += nums[right];

            while (windowSub >= target) {
                res = Math.min(res,right - left + 1);
                windowSub -= nums[left];
                left++;
                
            }

            
        }

        return res == Integer.MAX_VALUE ? 0 : res;
    }
}

File: Data Structures & Algorithms/is-palindrome/submission-3.cs
public class Solution
{
    public bool IsPalindrome(string s)
    {
        int left = 0;
        int right = s.Length - 1;

        string regulatedString = s.ToLowerInvariant();

        while (left < right)
        {
            while (left < right && !char.IsLetterOrDigit(regulatedString[left]))
            {
                left++;
            }

            while (left < right && !char.IsLetterOrDigit(regulatedString[right]))
            {
                right--;
            }

            if (regulatedString[left] != regulatedString[right])
            {
                return false;
            }

            left++;
            right--;
        }

        return true;
    }
}